### PR TITLE
[bitnami/harbor] Add service account for Harbor resources

### LINKED ISSUE #26870: [bitnami/harbor] service account for harbor resources
### Name and Version

bitnami/harbor:latest

### What is the problem this feature will solve?

current harbor helm chart doesn't create "serviceAccount" for any of harbor pods (excluding postgres & redis dependencies).
i guess relying of namespace default serviceAccount is not an good idea.

### What is the feature you are proposing to solve the problem?

a serviceAccount resource should be added in current chart. maybe one account for all pods. or maybe one for each component

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by longnbao**
### Description of the change

- Add a ServiceAccount for each Harbor service

### Benefits

- Using the default service account for all components in Kubernetes is not a security best practice.
- Add support to authenticate with Cloud Provider using Kubernetes workload identity and access through service account.

### Applicable issues

- fixes #26870

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by thang-dang-259 — APPROVED** [review]

**Comment by longnbao** [comment]
Hi @migruiz4 would you mind reviewing this PR? Thank you.

File: bitnami/harbor/templates/trivy/trivy-sts.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ spec:
         {{- end }}
     spec:
       {{- include "harbor.imagePullSecrets" . | nindent 6 }}
-      automountServiceAccountToken: {{ .Values.trivy.automountServiceAccountToken }}
+      automountServiceAccountToken: {{ .Values.trivy.serviceAccount.automountServiceAccountToken }}
```
Comment (migruiz4) [inline bitnami/harbor/templates/trivy/trivy-sts.yaml:42]: Please keep this line as it was:
```suggestion      
      automountServiceAccountToken: {{ .Values.trivy.automountServiceAccountToken }}
```

File: bitnami/harbor/templates/trivy/trivy-service-account.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+metadata:
+  name: {{ include "harbor.trivy.serviceAccountName" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: trivy
+  {{- if or .Values.trivy.serviceAccount.annotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" ( dict "values" ( list .Values.trivy.serviceAccount.annotations .Values.commonAnnotations ) "context" . ) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $) | nindent 4 }}
+  {{- end }}
+automountServiceAccountToken: {{ .Values.trivy.serviceAccount.automountServiceAccountToken }}
```
Comment (migruiz4) [inline bitnami/harbor/templates/trivy/trivy-service-account.yaml:18]: This one is okay 👍

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @longnbao,

Thank you very much for your contribution!

Could you please modify this PR to keep both `.Values.<component>.automountServiceAccountToken` and `.Values.<component>.serviceAccount.automountServiceAccountToken`?

Similar to how it is managed in other charts, we like to distinct between automountServiceAccountToken at pod level (.Values.<component>.automountServiceAccountToken) and ServiceAccount level (.Values.<component>.serviceAccount.automountServiceAccountToken).

Note: I have commented on Trivy STS and SA, but the same would apply to all Harbor components and values.yaml.

File: bitnami/harbor/templates/trivy/trivy-sts.yaml (hunk 1 above)
Comment (longnbao) [reply bitnami/harbor/templates/trivy/trivy-sts.yaml]: Hi @migruiz4 Updated, please take a look. Thank you.

**Comment by longnbao** [comment]
Hi @migruiz4 PTAL, thank you.

**Comment by longnbao** [comment]
Hi @migruiz4 would you mind reviewing this PR? Thank you.

**Review by migruiz4 — APPROVED** [review]
Hi @longnbao,

I'm sorry for the late response.

I have updated this PR with the following changes:
- Modified some values description to help differentiate between pod's automount setting and sa's automount setting.
- Bumped major version because of non-backward compatible change `<component>.serviceAccountName --> <component>.serviceAccount.name`. 
- Added upgrading notes in the README.md notifying of the non-backward compatible change.

**Pull request merged**